wp-content/plugins/hurumap/api.php

```diff
@@ -90,204 +44,89 @@ function register_routes()
-    }
 
-    $response->set_status(200);
-
-    return $response;
-}
+        $response = new WP_REST_Response($chart);
 
-function update_or_create_chart($json)
-{
-    global $wpdb;
+        $response->set_status(200);
 
-    if (!$wpdb->update(
-        "{$wpdb->base_prefix}hurumap_charts",
-        $json,
-        array('id' => $json['id'])
-    )) {
-        $wpdb->insert("{$wpdb->base_prefix}hurumap_charts", $json);
+        return $response;
     }
-}
 
-function update_or_create_charts($request)
-{
-    $json = $request->get_json_params();
-    if (is_array($json[0])) {
-        foreach ($json as $chart) {
-            update_or_create_chart($chart);
+    $posts = get_posts(array(
+        'posts_per_page'			=> -1,
+        'post_type'					=> 'hurumap-visual',
+        'post_status'				=> array('publish')
+    ));
+    
+    // Update $post_ids with a non false value.
+    $charts = array();
+    foreach( $posts as $post ) {
+        $chart = json_decode($post->post_content, true);
+        $chart['type'] = $post->post_excerpt;
+        if ($post->post_excerpt == 'hurumap') {
+            $chart["visual"]['queryAlias'] = "v{$chart['id']}";
+            if ($chart["stat"]) {
+                $chart["stat"]['queryAlias'] = "v{$chart['id']}";
+            }
         }
-    } else {
-        update_or_create_chart($json);
-    }
-    $response = new WP_REST_Response();
-    $response->set_status(200);
-
-    return $response;
-}
-
-function delete_charts($request)
-{
-    global $wpdb;
-
-    $json = $request->get_json_params();
-    $placeholders = implode(', ', array_fill(0, count($json), '%s'));
-    $success = $wpdb->query($wpdb->prepare("DELETE FROM {$wpdb->base_prefix}hurumap_charts WHERE id IN ({$placeholders})", $json));
-    $response = new WP_REST_Response($success);
-    $response->set_status(200);
-
-    return $response;
-}
-
-
-function update_or_create_section($json)
-{
-    global $wpdb;
-
-    if (!$wpdb->update(
-        "{$wpdb->base_prefix}chart_sections",
-        $json,
-        array('id' => $json['id'])
-    )) {
-        $wpdb->insert("{$wpdb->base_prefix}chart_sections", $json);
-        $res = array('ok' => true, 'id' => $wpdb->insert_id);
-    } else {
-        $res = array('ok' => true);
+        $charts[] = $chart;
     }
-    $response = new WP_REST_Response($res);
-    $response->set_status(200);
-}
 
-function update_or_create_sections($request)
-{
-    $json = $request->get_json_params();
-    if (is_array($json[0])) {
-        foreach ($json as $section) {
-            update_or_create_section($section);
-        }
-    } else {
-        update_or_create_section($json);
+    $posts = get_posts(array(
+        'posts_per_page'			=> -1,
+        'post_type'					=> 'hurumap-section',
+        'post_status'				=> array('publish')
+    ));
+    $sections = array();
+    foreach( $posts as $post ) {
```

Comment:
If no sections have been defined yet, `$posts` will be empty and empty array will be returned _even if we have charts..._

Reply:
Resolved with the `sectioned` query param